### DISCUSSION TITLE
Remote Data Portal Configuration

### DISCUSSION THREAD

**Discussion by kobi-shanti** [MarimerLLC/csla#3519] [Questions] [answered]
We are moving an existing application from a 2-tier to a 3-tier architecture to support Xamarin and reap some of the benefits of running an app server. 

So far we've managed to get our applications running in a mock 3-tier environment. To do this, we've configured CSLA to use the http proxy, and pointed our apps towards a 'remote' ASP.NET Core data portal server running on IIS Express (as per sample code). This is working fine when using IIS Express, but we're running into issues when trying to get the ASP.NET Core data portal server running on IIS. CSLA code that is executing fine when running on IIS Express is returning non-descriptive '...failed on server' messages when running on IIS. I've confirmed the data portal server is running on IIS by testing a basic get that isn't dependent on CSLA.

Is there any documentation available that covers the IIS configuration requirements for running a ASP.NET Core CSLA data portal server? Is there something obvious we've missed that would stop our CSLA data portal server running on IIS when it runs fine on IIS Express?

**Comment by hurcane** [comment]
Just a guess, but does the IIS server have the appropriate prerequisites installed. The appropriate hosting bundle for the target framework version is necessary, but not part of the default installation.

I would validate the server configuration per [Microsoft docs](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/?view=aspnetcore-7.0) . That link is for .NET 7.0.

**Reply by kobi-shanti** [reply]
Yes, we have configured IIS with the ASP.NET Core hosting bundle and confirmed that the ASP.Net Core 'server' is running on IIS by successfully testing the basic GET endpoint. The error is only thrown once the CSLA code is triggered by calling the data portal from the client.

**Comment by rockfordlhotka** [comment]
I haven't used IIS or IIS Express for many years. .NET Core (modern .NET) has its own built-in web server called Kestrel, which is lighter weight and also works on Linux, and I've been using that for the past many years.

**Reply by kobi-shanti** [reply]
Ok thanks Rocky, we will investigate this as an option if we can't get IIS to play nicely.

**Comment by kobi-shanti** [comment, marked as answer]
We found that the IIS app pool that our data portal server was using was trying to connect to our SQL DB as a user without appropriate authentication/authorization.
